### PR TITLE
Editing feature for Courses

### PR DESCRIPTION
**Pull request by kainguyen123-oss**
I added an editing feature for the courses.

### CODE REVIEW

File: portal/teacher.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def courses():
                 WHERE teacher_id = %s
             """, (g.user['id'],))
             courses = cur.fetchall()
+            print(courses)
```
Comment (kevinlueke) [inline portal/teacher.py]: This is for testing. Should be removed.

File: portal/templates/CourseCreation.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
++{% extends "layouts/base.html" %}
++
++{% block title %}Course Edit{% endblock %}
++
++{% block content %}
++<form method='post'>
```
Comment (kevinlueke) [inline portal/templates/CourseCreation.html]: Closing form tag is indented but this one is not. Indent this one.

File: portal/templates/CourseCreation.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
++  <label for="name">
++    <input type = "text" name = "course-name" id='name' />
++    </label>
```
Comment (kevinlueke) [inline portal/templates/CourseCreation.html]: Another indentation issue. Once these are fixed, might need to double check the rest of the indentation

File: portal/templates/CourseCreation.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
++    <label for="major">
++      <select name="class-major" id='major'>
++        <option value="cset" selected>CSET</option>
++        <option value="gen-ed">GEN-ED</option>
++        <option value="masonry">Masonry</option>
++        </select>
```
Comment (kevinlueke) [inline portal/templates/CourseCreation.html]: Indentation.

File: portal/templates/course-creation.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {% extends "layouts/base.html" %}
 
-{% block title %}Course creation{% endblock %}
+{% block title %}Course creation '{}'{% endblock %}
```
Comment (kevinlueke) [inline portal/templates/course-creation.html]: What is this doing with the curly braces?

File: portal/templates/edit.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <label for="major">
+      <select name="class-major" id='major'>
+        <option value="cset" selected>CSET</option>
+        <option value="gen-ed">GEN-ED</option>
+        <option value="masonry">Masonry</option>
+        </select>
+      </label>
+      <label for="code">
+        <input type = "text" name = "course-code" id='code' />
+        </label>
```
Comment (kevinlueke) [inline portal/templates/edit.html]: Indentation Issue.

File: portal/templates/edit.html [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        <label for="description">
+        <input type = "text" name = "course-description" id='description'/>
```
Comment (kevinlueke) [inline portal/templates/edit.html]: Indentation Issue

File: portal/templates/edit.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        </label>
+        <input type = "submit" />
```
Comment (kevinlueke) [inline portal/templates/edit.html]: The indentation in this file is getting indented in areas it shouldn't be and it becomes clear on this line.

File: tests/test_teacher.py [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import pytest
-from flask import g, session
+from flask import g, session, url_for, request
 from portal.db import get_db
-
+from urllib.parse import urlparse
```
Comment (kevinlueke) [inline tests/test_teacher.py]: Isn't used so not needed.

File: portal/teacher.py (hunk 1 above)
Comment (Steve-D-Eckles) [inline portal/teacher.py]: make sure to get rid of debug code like this

File: tests/test_teacher.py [hunk 10]
```diff
@@ -39,6 +39,31 @@ def test_creation(client, auth):
     #Asserts that it's switched over to the Course Selection
     assert 'http://localhost/teacher/courses' == response.headers['Location']
 
+def test_course_edit(client,auth,app):
+    auth.teacher_login()
+    #assert client.get('/teacher/courses/edit').status_code == 200
```
Comment (kevinlueke) [inline tests/test_teacher.py]: If these aren't being used they can be deleted.

Comment (kevinlueke) [reply tests/test_teacher.py]: These meaning the commented out code.

File: tests/test_teacher.py [hunk 11]
```diff
@@ -39,6 +39,31 @@ def test_creation(client, auth):
...
+            data={'code': 500, 'name': 'Ice-Cream', 'major': 'CSET',
+            'description': 'A basic introduction to Ice-Cream'}
+        )
+        with get_db() as con:
+            with con.cursor() as cur:
+                cur.execute("""
+                SELECT * FROM courses
+                WHERE id = %s
+                """,
+                (1, )
```
Comment (kevinlueke) [inline tests/test_teacher.py]: Get rid of that extra space.

**Review by kevinlueke — CHANGES_REQUESTED** [review]

File: tests/test_teacher.py [hunk 12]
```diff
@@ -39,6 +39,31 @@ def test_creation(client, auth):
...
+                )
+                result = cur.fetchone()
+                assert result['course_name'] == 'Ice-Cream'
+                assert res.headers['Location'] == 'http://localhost/teacher/courses'
```
Comment (Steve-D-Eckles) [inline tests/test_teacher.py:69]: include an empty line between the end of one function and the definition of the next for better readability

File: portal/templates/CourseCreation.html [hunk 13]
```diff
@@ -0,0 +1,25 @@
++{% extends "layouts/base.html" %}
```
Comment (Steve-D-Eckles) [inline portal/templates/CourseCreation.html]: this file was renamed 'course-creation.html' and does not need to be added back in with the old name

**Review by Steve-D-Eckles — CHANGES_REQUESTED** [review]

**Review by kevinlueke — APPROVED** [review]

**Review by Steve-D-Eckles — APPROVED** [review]

**Pull request merged**